src/namespace/kv.rs

```diff
@@ -0,0 +1,154 @@
+use crate::error::Result;
+use crate::namespace::prefix_internal;
+use crate::rpc::pb::etcdserverpb::request_op::Request;
+use crate::rpc::pb::etcdserverpb::response_op::Response;
+pub use crate::rpc::pb::etcdserverpb::{
+    Compare as PbCompare, DeleteRangeResponse as PbDeleteResponse, PutResponse as PbPutResponse,
+    RangeResponse as PbRangeResponse, TxnRequest as PbTxnRequest, TxnResponse as PbTxnResponse,
+};
+use crate::rpc::pb::etcdserverpb::{RequestOp, ResponseOp};
+use crate::{
+    take_mut, DeleteOptions, DeleteResponse, GetOptions, GetResponse, KvClient, PutOptions,
+    PutResponse, Txn, TxnResponse,
+};
+
+pub struct KvClientPrefix {
+    pfx: Vec<u8>,
+    kv: KvClient,
+}
+
+impl KvClientPrefix {
+    pub fn new(kv: KvClient, pfx: Vec<u8>) -> Self {
+        Self { pfx, kv }
+    }
+
+    pub async fn put(
+        &mut self,
+        key: impl Into<Vec<u8>>,
+        value: impl Into<Vec<u8>>,
+        options: Option<PutOptions>,
+    ) -> Result<PutResponse> {
+        let (key, _) = prefix_internal(&self.pfx, key.into(), vec![]);
+        let mut resp = self.kv.put(key, value, options).await?;
+        resp.take_mut_inner(|resp| self.strip_prefix_put_response(resp));
+        Ok(resp)
+    }
+
+    pub async fn get(
+        &mut self,
+        key: impl Into<Vec<u8>>,
+        options: Option<GetOptions>,
+    ) -> Result<GetResponse> {
+        let (key, _) = prefix_internal(&self.pfx, key.into(), vec![]);
+        let mut resp = self.kv.get(key, options).await?;
+        resp.take_mut_inner(|resp| self.strip_prefix_range_response(resp));
+        Ok(resp)
+    }
+
+    pub async fn delete(
+        &mut self,
+        key: impl Into<Vec<u8>>,
+        options: Option<DeleteOptions>,
+    ) -> Result<DeleteResponse> {
+        let (key, _) = prefix_internal(&self.pfx, key.into(), vec![]);
+        // TODO (@tisonkun) prefix range_end in options
```

Comment:
DeleteOptions make take_mut but the abstraction is strange a bit.

Others are implemented.